aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -218,19 +218,26 @@ impl<T: Transaction> CapturedReads<T> {
         }
     }
 
-    #[allow(dead_code)]
     pub(crate) fn capture_group_size(
         &mut self,
-        _state_key: T::Key,
-        _group_size: u64,
+        group_key: T::Key,
+        group_size: u64,
     ) -> anyhow::Result<()> {
-        unimplemented!("Group size capturing not implemented");
+        let group = self.group_reads.entry(group_key).or_default();
+
+        if let Some(recorded_size) = group.speculative_size {
+            if recorded_size != group_size {
+                bail!("Inconsistent recorded group size");
+            }
+        }
+
+        group.speculative_size = Some(group_size);
```

Comment:
Why is it called "speculative" size? Is it because the group writes could differ when the transactions are executed sequentially?

Also, why do we need to keep track of group size?

Reply:
Group size is needed for gas, if I am not mistaken (we charged proportionally to # bytes). It is speculative because of parallel execution as it is based on group members in MVHashMap

Reply:
indeed thanks @georgemitenkov, and great question @vusirikala - it all goes back to knowing the group size during transaction execution (particularly for gas). Of course everything is speculative at this point, and in a way, even before, it was "speculative" as the transaction execution was speculative, but the R/W conflict on the group would ensure it's correctness, while now it's recorded and validated on its own. I think I wanted to emphasize that this size is even more speculative than regular speculative, since it literally goes over the current latest group contents (involving multiple tags), and aggregates them (unlike reading a single latest speculative value). Let me rename it to collected_size and add a comment about this "extra-speculative" nature.